src/rust/engine/src/nodes.rs

```diff
@@ -644,6 +659,68 @@ impl From<Snapshot> for NodeKey {
   }
 }
 
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct DownloadedFile(Key);
+
+impl WrappedNode for DownloadedFile {
+  type Item = Arc<fs::Snapshot>;
+
+  fn run(self, context: Context) -> NodeFuture<Arc<fs::Snapshot>> {
+    let url_to_fetch = externs::project_str(&externs::val_for(&self.0), "url");
+
+    // Parse URL, get file
+    let url = try_future!(
+      Url::parse(&url_to_fetch).map_err(|err| throw(&format!("Error parsing URL: {}", err)))
+    );
+    let file_name = url
+      .path_segments()
+      .expect("TODO")
+      .last()
+      .expect("TODO")
+      .to_owned();
+
+    let tempdir = try_future!(
+      tempfile::TempDir::new()
+        .map_err(|err| throw(&format!("Error making tempdir to download file: {}", err)))
+    );
+
+    // TODO: Share a client which is pre-configured.
```

Comment:
A client should maybe be initialised and live on `Core` for sharing, as creating a client is non-trivial overhead

Reply:
+1

Although probably not a blocker for a first version.